### PR TITLE
release-23.1: kvprober: special case node-is-decommissioned errors

### PR DESCRIPTION
**Pull request by joshimhoff**
Release justification: `kvprober` is off by default and undocumented. It should only be used in CC prod as a result. Without this change, SRE is paged during decommission operations run in CC. 

Backport 1/1 commits from #104365.

/cc @cockroachdb/release

---

**kvprober: special case node-is-decommissioned errors**

kvprober runs on decommissioned node. In CC, this is generally fine, since
automation fully takes down nodes once they reach the decommissioned state. But
there is a brief period where a node is running and in the decommissioned
state, and we see kvprober errors in metrics during this period, as in below.
This sometimes leads to false positive kvprober pages in CC production.

‹rpc error: code = PermissionDenied desc = n1 was permanently removed from...

To be clear, the errors are not wrong per say. They just are expected to
happen, once a node is decommissioned.

This commit adds special handling for errors of the kind above, by doing a
substring match on the error string. To be exact, kvprober now logs such errors
at warning level and does not increment any error counters. This way, an
operation like decommissioning a node does not cause false positive kvprober
pages in CC production.

Fixes https://github.com/cockroachdb/cockroach/issues/104367

Release note: None, since kvprober is not used by customers. (It is not
documented.)

Co-authored-by: Josh Carp <carp@cockroachlabs.com>

### CODE REVIEW

**Comment by joshimhoff** [comment]
@nvanbenschoten, this is ready for review. Note that I have adjusted `errorIsExpectedDuringNormalOperation` to do a check with `string.Contains`, since `errorIsExpectedDuringNormalOperation` is not available in this branch. Note that I will run a manual test before actually merging, since we only have a unit test of the filtering in place.

**Review by nvb — APPROVED** [review]
:lgtm: if the manual testing you referenced was a success.

> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/106922#-:-N_ns3iCCAgjhuuFBnNO:b-6jvbpc)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @joshimhoff)

**Comment by joshimhoff** [comment]
TFTR! Local tests passed here too. Will merge once CI finishes.

**Pull request merged**
